public/app/features/alerting/unified/components/mute-timings/useMuteTimings.tsx

```diff
@@ -302,21 +260,24 @@ export const useUpdateMuteTiming = ({ alertmanager }: BaseAlertmanagerArgs) => {
 export const useDeleteMuteTiming = ({ alertmanager }: BaseAlertmanagerArgs) => {
   const useK8sApi = shouldUseK8sApi(alertmanager);
 
-  const dispatch = useDispatch();
+  const [updateConfiguration, _updateConfigurationRequestState] = useProduceNewAlertmanagerConfiguration();
   const [deleteGrafanaTimeInterval] = useDeleteNamespacedTimeIntervalMutation();
 
-  if (useK8sApi) {
-    return async ({ name }: { name: string }) => {
-      const namespace = getK8sNamespace();
-      return deleteGrafanaTimeInterval({
-        name,
-        namespace,
-        ioK8SApimachineryPkgApisMetaV1DeleteOptions: {},
-      }).unwrap();
-    };
-  }
+  const deleteFromAlertmangerAPI = useAsync(async (name: string) => {
```

Comment:
*deleteFromAlertmanagerAPI

Could we also make this back into `{ name }`? As we might want to pass in more options, and it might sometimes be an ID vs a Name? I originally had it that way to allow us the flexibility :+1:

Reply:
Definitely reasonable to allow for some API flexibility here 👍